app/controllers/users_controller.rb

```diff
@@ -0,0 +1,11 @@
+class UsersController < ApplicationController
+
+  def list
+    if current_user.admin?
```

Comment:
Rather than doing this in the controller action directly take a look at before filters:

https://guides.rubyonrails.org/action_controller_overview.html#filters

That way when we add more actions they will all be protected too